### ISSUE TITLE
Ember data with Octane @attr not working as expected

### ISSUE THREAD

**Issue by justkarenlo** [emberjs/data#6637]
### Reproduction
Context:
Working with Octane enabled for Ember version 3.13

I have a model that looks something like so:
```cli
import DS from 'ember-data';
const { Model, attr } = DS;

export default class SnackModel extends Model {
  @attr('string') cookieName;
  @attr('date') expirationDate;
  @attr recipe;
}
```

### Description
Within the route's model hook, we'll make a fetch to retrieve the snack model data from the API. When I look at the model that gets generated, only the attributes with transformers exist in the snack model context. In this example, `recipe` would not be returned when we log the model, but cookieName and expirationDate would exist.

If I revert back to using class extends syntax, the full model is generated correctly. Example when I revert back to old way of defining a model (all properties exist and are available again):
```cli
import DS from 'ember-data';
const { Model, attr } = DS;

export default Model.extend({
  cookieName: attr('string'),
  expirationDate: attr('date'),
  recipe: attr(),
});
```

Example of what the API is returning:
```cli
{
  cookieName: 'Chocolate Chip',
  expirationDate: 1503039600000,
  recipe: {
    sugar: '1 cup',
    flour: '2 cups'
  },
}
```

### Versions
```cli
enterprise-customer-exp-web@0.0.0-productspec /Users/klo/workspace/enterprise-customer-exp-web_trunk
└── ember-source@3.13.2 
enterprise-customer-exp-web@0.0.0-productspec /Users/klo/workspace/enterprise-customer-exp-web_trunk
└── ember-cli@3.13.1 

enterprise-customer-exp-web@0.0.0-productspec /Users/klo/workspace/enterprise-customer-exp-web_trunk
└── ember-data@3.8.0 
```

**Comment by pzuraq** [comment]
Thanks for this detailed bug report! The issue you're experiencing is because older versions of Ember Data did not support using `@attr` without parenthesis. You would have to use parens, even if no values were passed:

```js
export default class SnackModel extends Model {
  @attr('string') cookieName;
  @attr('date') expirationDate;
  @attr() recipe;
}
```

This was fixed in 3.13. However, we just updated [the decorators polyfill]( https://github.com/pzuraq/ember-decorators-polyfill) to polyfill support for this, so if you install it it should start working!

**Comment by justkarenlo** [comment]
Thanks for the quick response.

You stated that this was fixed in 3.13 and the polyfill "addon is not needed in Ember 3.10+", but our application ember-source is already using 3.13.2. When you said 3.13, are you referring to the ember-data version?

**Comment by pzuraq** [comment]
Correct, this was an issue with Ember Data, not Ember, so you need Ember Data 3.13 (or the polyfill) to get the behavior you're after.

**Comment by runspired** [comment]
As @pzuraq stated the reason this does not work is due to the use of EmberData 3.8. If the issue persists after upgrading EmberData to 3.13 or when using the polyfill then let's reopen.

**Comment by justkarenlo** [comment]
I can confirm that this issue was resolved after upgrading EmberData to 3.13. Thanks for the quick responses!
